### PR TITLE
Release `v0.38.0-rc1`

### PR DESCRIPTION
**Pull request by sergio-mena**
Changes in preparation for the first release candidate (RC) of v0.38.0:

- [CHANGELOG rendered](https://github.com/cometbft/cometbft/blob/release/v0.38.0-rc1/CHANGELOG.md)
- [Manual E2E run](https://github.com/cometbft/cometbft/actions/runs/5180113394) (for the `release/v0.38.0-rc1` branch)

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by hvanz — APPROVED** [review]
LGTM

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
             go.sum
       - uses: golangci/golangci-lint-action@v3
         with:
-          version: latest
+          version: v1.52.0
```
Comment (thanethomson) [inline .github/workflows/lint.yml:34]: Could we please avoid pinning golangci-lint to a specific version?

I've submitted #922 to fix the lints on the `v0.38.x` branch.

File: Makefile [hunk 2]
```diff
@@ -231,7 +231,7 @@ format:
 
 lint:
 	@echo "--> Running linter"
-	@go run github.com/golangci/golangci-lint/cmd/golangci-lint@latest run
+	@go run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.52.0 run
```
Comment (thanethomson) [inline Makefile:234]: Same as previous comment.

**Review by thanethomson — APPROVED** [review]
One request re: the golangci-lint changes, but otherwise LGTM, thanks @sergio-mena!

File: .github/workflows/lint.yml (hunk 1 above)
Comment (sergio-mena) [reply .github/workflows/lint.yml:34]: Ok, will revert my commit and wait for #922 to be merged

**Pull request merged**